heat/core/linalg/svdtools.py

```diff
@@ -0,0 +1,500 @@
+"""
+distributed hierarchical SVD
+"""
+import numpy as np
+import collections
+import torch
+from typing import Type, Callable, Dict, Any, TypeVar, Union, Tuple, Optional
+
+from ..communication import MPICommunication
+from ..dndarray import DNDarray
+from .. import factories
+from .. import types
+from ..linalg import matmul, vector_norm
+from ..indexing import where
+from ..random import randn
+
+from ..manipulations import vstack, hstack, diag, balance
+
+from .. import statistics
+from math import log, ceil, floor, sqrt
+
+
+__all__ = ["hsvd_rank", "hsvd_rtol", "hsvd"]
+
+
+#########################################################################################
+# user-friendly versions of hSVD
+#########################################################################################
+
+
+def hsvd_rank(
+    A: DNDarray,
+    maxrank: int,
+    full: bool = False,
+    maxmergedim: Optional[int] = None,
+    safetyshift: int = 5,
+    silent: bool = True,
```

Comment:
I may have misunderstood this, but the documentation @mtar refered to last week (https://docs.python.org/3/library/typing.html#typing.Optional) says: "An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional" 

In particular the code example 
```
def foo(arg: int = 0) -> None:
    ...
```
given there seems to support declarations like `full: bool = False` instead of the proposed change `full: Optional[bool] = False` or did I understand your change request in a wrong way?